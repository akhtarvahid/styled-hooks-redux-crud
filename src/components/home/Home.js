import React, { useEffect, useState } from "react";
import { useSelector, useDispatch } from "react-redux";
import { toast } from "react-toastify";
import "react-toastify/dist/ReactToastify.css";

import { v4 as uuidv4 } from 'uuid';
import ListItem from "../listItem/ListItem";
import {
  getItems, addItem, updateItem,
  deleteItem, checkItem, allChecked,
  deleteItems
} from "../../actions/crudActions";
import Header from "../header/Header";
import { Jumbotron, Container } from "react-bootstrap";
import Update from "../update/Update";
import Create from "../create/Create";
import Spinner from "../loader/Loader";
import './home.scss';

toast.configure({
  autoClose: 8000,
  draggable: false,
  position: toast.POSITION.BOTTOM_RIGHT
});
function Home() {
  const [state, setState] = useState({
    title: "",
    description: "",
    loading: false,
    checkedArr: []
  });
  const [status, setStatus] = useState({
    edit: false,
    add: false
  });
  const dispatch = useDispatch();
  const { lists } = useSelector(state => state.lists);
  const list = useSelector(state => state);

  useEffect(() => {
    dispatch(getItems());
  }, [dispatch]);

  const handleChange = e => {
    const { name, value } = e.target;
    setState({ ...state, [name]: value });
  };
  const submitCreateForm = e => {
    const { title, description } = state;

    e.preventDefault();
    let obj = {
      id: uuidv4(),
      title,
      description
    };
    if (state.title !== '' && state.description !== '') {
      setState({ ...state, loading: true });
      dispatch(addItem(obj));
      setTimeout(() => {
        if (!list.loading) {
          setState({ ...state, title: "", description: "", loading: false });
          toast.success("Added Successfully");
        }
      }, 1000);
    } else {
      toast.warning("Title & Description is Empty");
    }
    setStatus({ ...status, add: false });
  }
  const submitUpdateForm = e => {
    e.preventDefault();

    setState({ ...state, loading: true });
    dispatch(updateItem(state));
    setTimeout(() => {
      if (!list.loading) {
        setState({ ...state, title: "", description: "", loading: false });
        toast.success("Successfully Updated");
      }
    }, 1000);
    setStatus({ ...status, edit: false });
  };
  const onHandleEdit = (list) => {
    const { title, description, id } = list;
    setState({
      ...state,
      id: id,
      title: title,
      description: description
    });
    setStatus({ ...status, edit: true });
  };
  const addHandle = () => {
    setStatus({ ...status, add: true });
  }
  const onDelete = (list) => {
    dispatch(deleteItem(list));
    toast.success(`(${list.title}) deleted Successfully`);
  }
  const handleClose = () => {
    setStatus({ ...status, add: false, edit: false });
    setState({ ...state, title: "", description: "", loading: false });
  }

  const onChecked = (e) => {
    dispatch(checkItem(e.target.value, e.target.checked))
  }
  const handleAllChecked = (e) => {
    dispatch(allChecked(e.target.checked));
  }
  const deleteAll = () => {
    dispatch(deleteItems());
  }
  let AnyChecked = lists.some(list => list.completed === true);

  return (
    <div>
      <Header addHandle={addHandle} />
      {state.loading && <Spinner />}
      <div>
        <Container>
          <Jumbotron>
            <ListItem lists={lists}
              onHandleEdit={onHandleEdit}
              loading={true}
              onDelete={onDelete}
              deleteAll={deleteAll}
              onChecked={onChecked}
              AnyChecked={AnyChecked}
              handleAllChecked={handleAllChecked}
            />
          </Jumbotron>
        </Container>
      </div>
      {status.edit ?
        <Update
          status={status.edit}
          handleClose={handleClose}
          handleChange={handleChange}
          submitUpdateForm={submitUpdateForm}
          state={state}
        /> :
        <Create
          status={status.add}
          handleClose={handleClose}
          handleChange={handleChange}
          submitCreateForm={submitCreateForm}
          state={state}
        />
      }
    </div>
  )}

export default Home;
