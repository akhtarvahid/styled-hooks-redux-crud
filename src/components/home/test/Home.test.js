import React from 'react';
import renderer from 'react-test-renderer';
import Home from '../Home';
import { createStore } from "redux";
import { Provider } from "react-redux";
import rootReducer from '../../../reducers';
test('should render Home Component', () => {
    const HomeWrapper = () => {
        const store = createStore(rootReducer);
        return (
            <Provider store={store}>
                <Home />
            </Provider>
        )
    }

    const tree = renderer
        .create(<HomeWrapper />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});
