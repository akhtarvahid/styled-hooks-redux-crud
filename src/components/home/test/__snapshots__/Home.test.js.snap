// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Home Component 1`] = `
<div>
  <div
    className="menu-header"
  >
    <div
      className="menu-header__menu-inner"
    >
      <div>
        <img
          alt="logo"
          className="d-inline-block align-top"
          height="50"
          src="logo.gif"
          width="80"
        />
        <div>
          REACT CRUD
        </div>
      </div>
      <button
        onClick={[Function]}
      >
        Add
      </button>
    </div>
  </div>
  <div>
    <div
      className="container"
    >
      <div
        className="jumbotron"
      >
        <div
          className="table-responsive"
        >
          <table
            className="table table"
            style={
              Object {
                "display": false,
              }
            }
          >
            <thead>
              <tr>
                <th>
                  <svg
                    fill="currentColor"
                    height="1em"
                    stroke="currentColor"
                    strokeWidth="0"
                    style={
                      Object {
                        "color": undefined,
                      }
                    }
                    viewBox="0 0 448 512"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M448 360V24c0-13.3-10.7-24-24-24H96C43 0 0 43 0 96v320c0 53 43 96 96 96h328c13.3 0 24-10.7 24-24v-16c0-7.5-3.5-14.3-8.9-18.7-4.2-15.4-4.2-59.3 0-74.7 5.4-4.3 8.9-11.1 8.9-18.6zM128 134c0-3.3 2.7-6 6-6h212c3.3 0 6 2.7 6 6v20c0 3.3-2.7 6-6 6H134c-3.3 0-6-2.7-6-6v-20zm0 64c0-3.3 2.7-6 6-6h212c3.3 0 6 2.7 6 6v20c0 3.3-2.7 6-6 6H134c-3.3 0-6-2.7-6-6v-20zm253.4 250H96c-17.7 0-32-14.3-32-32 0-17.6 14.4-32 32-32h285.4c-1.9 17.1-1.9 46.9 0 64z"
                    />
                  </svg>
                </th>
                <th>
                  Title
                </th>
                <th
                  className="remove"
                >
                  Description
                </th>
                <th>
                  <span
                    className="action"
                  >
                    Action
                  </span>
                  <span
                    className="check-all"
                  >
                    <input
                      checked={false}
                      onChange={[Function]}
                      type="checkbox"
                    />
                  </span>
                </th>
              </tr>
            </thead>
            <tbody>
              <tr
                className=""
              >
                <td>
                  <svg
                    fill="currentColor"
                    height="1em"
                    stroke="currentColor"
                    strokeWidth="0"
                    style={
                      Object {
                        "color": undefined,
                      }
                    }
                    viewBox="0 0 448 512"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M448 360V24c0-13.3-10.7-24-24-24H96C43 0 0 43 0 96v320c0 53 43 96 96 96h328c13.3 0 24-10.7 24-24v-16c0-7.5-3.5-14.3-8.9-18.7-4.2-15.4-4.2-59.3 0-74.7 5.4-4.3 8.9-11.1 8.9-18.6zM128 134c0-3.3 2.7-6 6-6h212c3.3 0 6 2.7 6 6v20c0 3.3-2.7 6-6 6H134c-3.3 0-6-2.7-6-6v-20zm0 64c0-3.3 2.7-6 6-6h212c3.3 0 6 2.7 6 6v20c0 3.3-2.7 6-6 6H134c-3.3 0-6-2.7-6-6v-20zm253.4 250H96c-17.7 0-32-14.3-32-32 0-17.6 14.4-32 32-32h285.4c-1.9 17.1-1.9 46.9 0 64z"
                    />
                  </svg>
                </td>
                <td>
                  <span>
                    JavaScript
                  </span>
                </td>
                <td
                  className="remove"
                >
                  JavaScript
                </td>
                <td
                  className="table_edit"
                >
                  <div>
                    <svg
                      fill="currentColor"
                      height="1em"
                      onClick={[Function]}
                      stroke="currentColor"
                      strokeWidth="0"
                      style={
                        Object {
                          "color": undefined,
                        }
                      }
                      viewBox="0 0 576 512"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M402.6 83.2l90.2 90.2c3.8 3.8 3.8 10 0 13.8L274.4 405.6l-92.8 10.3c-12.4 1.4-22.9-9.1-21.5-21.5l10.3-92.8L388.8 83.2c3.8-3.8 10-3.8 13.8 0zm162-22.9l-48.8-48.8c-15.2-15.2-39.9-15.2-55.2 0l-35.4 35.4c-3.8 3.8-3.8 10 0 13.8l90.2 90.2c3.8 3.8 10 3.8 13.8 0l35.4-35.4c15.2-15.3 15.2-40 0-55.2zM384 346.2V448H64V128h229.8c3.2 0 6.2-1.3 8.5-3.5l40-40c7.6-7.6 2.2-20.5-8.5-20.5H48C21.5 64 0 85.5 0 112v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V306.2c0-10.7-12.9-16-20.5-8.5l-40 40c-2.2 2.3-3.5 5.3-3.5 8.5z"
                      />
                    </svg>
                    <input
                      checked={false}
                      onChange={[Function]}
                      type="checkbox"
                      value="JavaScript"
                    />
                  </div>
                </td>
              </tr>
              <tr
                className=""
              >
                <td>
                  <svg
                    fill="currentColor"
                    height="1em"
                    stroke="currentColor"
                    strokeWidth="0"
                    style={
                      Object {
                        "color": undefined,
                      }
                    }
                    viewBox="0 0 448 512"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M448 360V24c0-13.3-10.7-24-24-24H96C43 0 0 43 0 96v320c0 53 43 96 96 96h328c13.3 0 24-10.7 24-24v-16c0-7.5-3.5-14.3-8.9-18.7-4.2-15.4-4.2-59.3 0-74.7 5.4-4.3 8.9-11.1 8.9-18.6zM128 134c0-3.3 2.7-6 6-6h212c3.3 0 6 2.7 6 6v20c0 3.3-2.7 6-6 6H134c-3.3 0-6-2.7-6-6v-20zm0 64c0-3.3 2.7-6 6-6h212c3.3 0 6 2.7 6 6v20c0 3.3-2.7 6-6 6H134c-3.3 0-6-2.7-6-6v-20zm253.4 250H96c-17.7 0-32-14.3-32-32 0-17.6 14.4-32 32-32h285.4c-1.9 17.1-1.9 46.9 0 64z"
                    />
                  </svg>
                </td>
                <td>
                  <span>
                    Sass
                  </span>
                </td>
                <td
                  className="remove"
                >
                  Learn from 
                </td>
                <td
                  className="table_edit"
                >
                  <div>
                    <svg
                      fill="currentColor"
                      height="1em"
                      onClick={[Function]}
                      stroke="currentColor"
                      strokeWidth="0"
                      style={
                        Object {
                          "color": undefined,
                        }
                      }
                      viewBox="0 0 576 512"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M402.6 83.2l90.2 90.2c3.8 3.8 3.8 10 0 13.8L274.4 405.6l-92.8 10.3c-12.4 1.4-22.9-9.1-21.5-21.5l10.3-92.8L388.8 83.2c3.8-3.8 10-3.8 13.8 0zm162-22.9l-48.8-48.8c-15.2-15.2-39.9-15.2-55.2 0l-35.4 35.4c-3.8 3.8-3.8 10 0 13.8l90.2 90.2c3.8 3.8 10 3.8 13.8 0l35.4-35.4c15.2-15.3 15.2-40 0-55.2zM384 346.2V448H64V128h229.8c3.2 0 6.2-1.3 8.5-3.5l40-40c7.6-7.6 2.2-20.5-8.5-20.5H48C21.5 64 0 85.5 0 112v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V306.2c0-10.7-12.9-16-20.5-8.5l-40 40c-2.2 2.3-3.5 5.3-3.5 8.5z"
                      />
                    </svg>
                    <input
                      checked={false}
                      onChange={[Function]}
                      type="checkbox"
                      value="Sass"
                    />
                  </div>
                </td>
              </tr>
              <tr
                className=""
              >
                <td>
                  <svg
                    fill="currentColor"
                    height="1em"
                    stroke="currentColor"
                    strokeWidth="0"
                    style={
                      Object {
                        "color": undefined,
                      }
                    }
                    viewBox="0 0 448 512"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M448 360V24c0-13.3-10.7-24-24-24H96C43 0 0 43 0 96v320c0 53 43 96 96 96h328c13.3 0 24-10.7 24-24v-16c0-7.5-3.5-14.3-8.9-18.7-4.2-15.4-4.2-59.3 0-74.7 5.4-4.3 8.9-11.1 8.9-18.6zM128 134c0-3.3 2.7-6 6-6h212c3.3 0 6 2.7 6 6v20c0 3.3-2.7 6-6 6H134c-3.3 0-6-2.7-6-6v-20zm0 64c0-3.3 2.7-6 6-6h212c3.3 0 6 2.7 6 6v20c0 3.3-2.7 6-6 6H134c-3.3 0-6-2.7-6-6v-20zm253.4 250H96c-17.7 0-32-14.3-32-32 0-17.6 14.4-32 32-32h285.4c-1.9 17.1-1.9 46.9 0 64z"
                    />
                  </svg>
                </td>
                <td>
                  <span>
                    React
                  </span>
                </td>
                <td
                  className="remove"
                >
                  Learn from react.js
                </td>
                <td
                  className="table_edit"
                >
                  <div>
                    <svg
                      fill="currentColor"
                      height="1em"
                      onClick={[Function]}
                      stroke="currentColor"
                      strokeWidth="0"
                      style={
                        Object {
                          "color": undefined,
                        }
                      }
                      viewBox="0 0 576 512"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M402.6 83.2l90.2 90.2c3.8 3.8 3.8 10 0 13.8L274.4 405.6l-92.8 10.3c-12.4 1.4-22.9-9.1-21.5-21.5l10.3-92.8L388.8 83.2c3.8-3.8 10-3.8 13.8 0zm162-22.9l-48.8-48.8c-15.2-15.2-39.9-15.2-55.2 0l-35.4 35.4c-3.8 3.8-3.8 10 0 13.8l90.2 90.2c3.8 3.8 10 3.8 13.8 0l35.4-35.4c15.2-15.3 15.2-40 0-55.2zM384 346.2V448H64V128h229.8c3.2 0 6.2-1.3 8.5-3.5l40-40c7.6-7.6 2.2-20.5-8.5-20.5H48C21.5 64 0 85.5 0 112v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V306.2c0-10.7-12.9-16-20.5-8.5l-40 40c-2.2 2.3-3.5 5.3-3.5 8.5z"
                      />
                    </svg>
                    <input
                      checked={false}
                      onChange={[Function]}
                      type="checkbox"
                      value="React"
                    />
                  </div>
                </td>
              </tr>
              <tr
                className=""
              >
                <td>
                  <svg
                    fill="currentColor"
                    height="1em"
                    stroke="currentColor"
                    strokeWidth="0"
                    style={
                      Object {
                        "color": undefined,
                      }
                    }
                    viewBox="0 0 448 512"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M448 360V24c0-13.3-10.7-24-24-24H96C43 0 0 43 0 96v320c0 53 43 96 96 96h328c13.3 0 24-10.7 24-24v-16c0-7.5-3.5-14.3-8.9-18.7-4.2-15.4-4.2-59.3 0-74.7 5.4-4.3 8.9-11.1 8.9-18.6zM128 134c0-3.3 2.7-6 6-6h212c3.3 0 6 2.7 6 6v20c0 3.3-2.7 6-6 6H134c-3.3 0-6-2.7-6-6v-20zm0 64c0-3.3 2.7-6 6-6h212c3.3 0 6 2.7 6 6v20c0 3.3-2.7 6-6 6H134c-3.3 0-6-2.7-6-6v-20zm253.4 250H96c-17.7 0-32-14.3-32-32 0-17.6 14.4-32 32-32h285.4c-1.9 17.1-1.9 46.9 0 64z"
                    />
                  </svg>
                </td>
                <td>
                  <span>
                    Node
                  </span>
                </td>
                <td
                  className="remove"
                >
                  Nodejs documentation
                </td>
                <td
                  className="table_edit"
                >
                  <div>
                    <svg
                      fill="currentColor"
                      height="1em"
                      onClick={[Function]}
                      stroke="currentColor"
                      strokeWidth="0"
                      style={
                        Object {
                          "color": undefined,
                        }
                      }
                      viewBox="0 0 576 512"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M402.6 83.2l90.2 90.2c3.8 3.8 3.8 10 0 13.8L274.4 405.6l-92.8 10.3c-12.4 1.4-22.9-9.1-21.5-21.5l10.3-92.8L388.8 83.2c3.8-3.8 10-3.8 13.8 0zm162-22.9l-48.8-48.8c-15.2-15.2-39.9-15.2-55.2 0l-35.4 35.4c-3.8 3.8-3.8 10 0 13.8l90.2 90.2c3.8 3.8 10 3.8 13.8 0l35.4-35.4c15.2-15.3 15.2-40 0-55.2zM384 346.2V448H64V128h229.8c3.2 0 6.2-1.3 8.5-3.5l40-40c7.6-7.6 2.2-20.5-8.5-20.5H48C21.5 64 0 85.5 0 112v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V306.2c0-10.7-12.9-16-20.5-8.5l-40 40c-2.2 2.3-3.5 5.3-3.5 8.5z"
                      />
                    </svg>
                    <input
                      checked={false}
                      onChange={[Function]}
                      type="checkbox"
                      value="Node"
                    />
                  </div>
                </td>
              </tr>
              <tr
                className=""
              >
                <td>
                  <svg
                    fill="currentColor"
                    height="1em"
                    stroke="currentColor"
                    strokeWidth="0"
                    style={
                      Object {
                        "color": undefined,
                      }
                    }
                    viewBox="0 0 448 512"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M448 360V24c0-13.3-10.7-24-24-24H96C43 0 0 43 0 96v320c0 53 43 96 96 96h328c13.3 0 24-10.7 24-24v-16c0-7.5-3.5-14.3-8.9-18.7-4.2-15.4-4.2-59.3 0-74.7 5.4-4.3 8.9-11.1 8.9-18.6zM128 134c0-3.3 2.7-6 6-6h212c3.3 0 6 2.7 6 6v20c0 3.3-2.7 6-6 6H134c-3.3 0-6-2.7-6-6v-20zm0 64c0-3.3 2.7-6 6-6h212c3.3 0 6 2.7 6 6v20c0 3.3-2.7 6-6 6H134c-3.3 0-6-2.7-6-6v-20zm253.4 250H96c-17.7 0-32-14.3-32-32 0-17.6 14.4-32 32-32h285.4c-1.9 17.1-1.9 46.9 0 64z"
                    />
                  </svg>
                </td>
                <td>
                  <span>
                    EcmaScript
                  </span>
                </td>
                <td
                  className="remove"
                >
                  Learn from es6.org
                </td>
                <td
                  className="table_edit"
                >
                  <div>
                    <svg
                      fill="currentColor"
                      height="1em"
                      onClick={[Function]}
                      stroke="currentColor"
                      strokeWidth="0"
                      style={
                        Object {
                          "color": undefined,
                        }
                      }
                      viewBox="0 0 576 512"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M402.6 83.2l90.2 90.2c3.8 3.8 3.8 10 0 13.8L274.4 405.6l-92.8 10.3c-12.4 1.4-22.9-9.1-21.5-21.5l10.3-92.8L388.8 83.2c3.8-3.8 10-3.8 13.8 0zm162-22.9l-48.8-48.8c-15.2-15.2-39.9-15.2-55.2 0l-35.4 35.4c-3.8 3.8-3.8 10 0 13.8l90.2 90.2c3.8 3.8 10 3.8 13.8 0l35.4-35.4c15.2-15.3 15.2-40 0-55.2zM384 346.2V448H64V128h229.8c3.2 0 6.2-1.3 8.5-3.5l40-40c7.6-7.6 2.2-20.5-8.5-20.5H48C21.5 64 0 85.5 0 112v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V306.2c0-10.7-12.9-16-20.5-8.5l-40 40c-2.2 2.3-3.5 5.3-3.5 8.5z"
                      />
                    </svg>
                    <input
                      checked={false}
                      onChange={[Function]}
                      type="checkbox"
                      value="EcmaScript"
                    />
                  </div>
                </td>
              </tr>
              <tr
                className=""
              >
                <td>
                  <svg
                    fill="currentColor"
                    height="1em"
                    stroke="currentColor"
                    strokeWidth="0"
                    style={
                      Object {
                        "color": undefined,
                      }
                    }
                    viewBox="0 0 448 512"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M448 360V24c0-13.3-10.7-24-24-24H96C43 0 0 43 0 96v320c0 53 43 96 96 96h328c13.3 0 24-10.7 24-24v-16c0-7.5-3.5-14.3-8.9-18.7-4.2-15.4-4.2-59.3 0-74.7 5.4-4.3 8.9-11.1 8.9-18.6zM128 134c0-3.3 2.7-6 6-6h212c3.3 0 6 2.7 6 6v20c0 3.3-2.7 6-6 6H134c-3.3 0-6-2.7-6-6v-20zm0 64c0-3.3 2.7-6 6-6h212c3.3 0 6 2.7 6 6v20c0 3.3-2.7 6-6 6H134c-3.3 0-6-2.7-6-6v-20zm253.4 250H96c-17.7 0-32-14.3-32-32 0-17.6 14.4-32 32-32h285.4c-1.9 17.1-1.9 46.9 0 64z"
                    />
                  </svg>
                </td>
                <td>
                  <span>
                    Angular
                  </span>
                </td>
                <td
                  className="remove"
                >
                  One framework Mobile...
                </td>
                <td
                  className="table_edit"
                >
                  <div>
                    <svg
                      fill="currentColor"
                      height="1em"
                      onClick={[Function]}
                      stroke="currentColor"
                      strokeWidth="0"
                      style={
                        Object {
                          "color": undefined,
                        }
                      }
                      viewBox="0 0 576 512"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M402.6 83.2l90.2 90.2c3.8 3.8 3.8 10 0 13.8L274.4 405.6l-92.8 10.3c-12.4 1.4-22.9-9.1-21.5-21.5l10.3-92.8L388.8 83.2c3.8-3.8 10-3.8 13.8 0zm162-22.9l-48.8-48.8c-15.2-15.2-39.9-15.2-55.2 0l-35.4 35.4c-3.8 3.8-3.8 10 0 13.8l90.2 90.2c3.8 3.8 10 3.8 13.8 0l35.4-35.4c15.2-15.3 15.2-40 0-55.2zM384 346.2V448H64V128h229.8c3.2 0 6.2-1.3 8.5-3.5l40-40c7.6-7.6 2.2-20.5-8.5-20.5H48C21.5 64 0 85.5 0 112v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V306.2c0-10.7-12.9-16-20.5-8.5l-40 40c-2.2 2.3-3.5 5.3-3.5 8.5z"
                      />
                    </svg>
                    <input
                      checked={false}
                      onChange={[Function]}
                      type="checkbox"
                      value="Angular"
                    />
                  </div>
                </td>
              </tr>
            </tbody>
          </table>
        </div>
        <div
          className="table_no-content"
        />
      </div>
    </div>
  </div>
</div>
`;
